external-crates/move/crates/move-analyzer/editors/code/src/context.ts

```diff
@@ -177,25 +203,176 @@ export class Context {
      */
     registerOnDidChangeConfiguration(): void {
         vscode.workspace.onDidChangeConfiguration(async event => {
-            const changed = event.affectsConfiguration('move.lint') ||
-                            event.affectsConfiguration('move.inlay-hints.type');
-            if (changed) {
-                const newLintLevel = lint();
-                const newInlayHintsType = inlayHintsForType();
-                if (this.lintLevel !== newLintLevel ||
-                    this.inlayHintsType !== newInlayHintsType) {
-                    this.lintLevel = newLintLevel;
-                    this.inlayHintsType = newInlayHintsType;
-                    try {
-                        await this.stopClient();
+
+            const server_path_conf = MOVE_CONF_NAME.concat('.').concat(SERVER_PATH_OPT);
+            const sui_path_conf = MOVE_CONF_NAME.concat('.').concat(SUI_PATH_OPT);
+            const lint_conf = MOVE_CONF_NAME.concat('.').concat(LINT_OPT);
+            const type_hints_conf = MOVE_CONF_NAME.concat('.').concat(TYPE_HINTS_OPT);
+
+            const optionsChanged = event.affectsConfiguration(lint_conf) ||
+                event.affectsConfiguration(type_hints_conf);
+            const pathsChanged = event.affectsConfiguration(server_path_conf) ||
+                event.affectsConfiguration(sui_path_conf);
+
+            if (optionsChanged || pathsChanged) {
+                this.configuration = new Configuration();
+                log.info(`configuration: ${this.configuration.toString()}`);
+
+                this.lintLevel = this.configuration.lint;
+                this.inlayHintsType = this.configuration.inlayHintsForType;
+                try {
+                    await this.stopClient();
+                        if (pathsChanged) {
+                            await this.installServerBinary(this.extensionContext);
+                        }
                         await this.startClient();
-                    } catch (err) {
-                        // Handle error
-                        log.info(String(err));
-                    }
+                } catch (err) {
+                    // Handle error
+                    log.info(String(err));
                 }
             }
         });
 
     }
+
+    async installBundledBinary(bundledServerPath: vscode.Uri): Promise<void> {
```

Comment:
Would some logs make sense here? For example, removing the existing binary might be useful... 

On another issue, this would fail if there is no disk space, right?

Reply:
> Would some logs make sense here? For example, removing the existing binary might be useful...

Added additional info to the logs

> On another issue, this would fail if there is no disk space, right?

Indeed, but presumably all sorts of things will start failing on the system at this point so I'd think we don't have to worry about handling this case explicitly here